src/lib/pages/frontsite/Documents/DocumentDetails/DocumentCirculation/DocumentEItems.js

```diff
@@ -14,10 +15,42 @@ export class DocumentEItems extends Component {
   };
 
   onClickEItemRequestLink = () => {
+    const {
+      documentDetails: { metadata },
+    } = this.props;
     const medium = 'DIGITAL';
+
+    const formData = {};
+
+    const authors = _get(metadata, 'authors[0].full_name');
+    const title = _get(metadata, 'title');
+    const publicationYear = _get(metadata, 'publication_year');
+    const publisher = _get(metadata, 'imprint.publisher');
+    const isbn = _get(metadata, 'identifiers[0].value');
+
+    if (authors) {
+      formData['authors'] = authors;
+    }
+
+    if (title) {
+      formData['title'] = title;
+    }
+
+    if (publicationYear) {
+      formData['publication_year'] = publicationYear;
+    }
+
+    if (publisher) {
+      formData['publisher'] = publisher;
+    }
+
+    if (isbn) {
+      formData['isbn'] = isbn;
+    }
+
     return {
       pathname: FrontSiteRoutes.documentRequestForm,
-      state: { medium },
+      state: { medium, formData },
```

Comment:
shouldn't the medium be a part of the form data ? is there any reason of separating them ?

Reply:
good catch!